### PR TITLE
GDB-4546 Provides popup to update single cell

### PR DESCRIPTION
**Pull request by teodossidossev**
Provides Factory pattern to operate with concrete objects without exposing the logic behind, using an interface.
Does update properties of a single mapping. Do not support type changes.

### CODE REVIEW

File: src/app/main/mapper/iteration/iteration.component.ts [hunk 1]
```diff
@@ -1,33 +1,93 @@
...
   templateUrl: './iteration.component.html',
   styleUrls: ['./iteration.component.scss'],
 })
 export class IterationComponent implements OnInit {
-  @Input() subject: SubjectMappingImpl | ValueMappingImpl;
+  @Input() mapping: MappingDefinitionImpl;
 
-  constructor(private modelManagementService: ModelManagementService) {
+  SUBJECT = SUBJECT_SELECTOR;
+  PREDICATE =PREDICATE_SELECTOR;
```
Comment (desislava-hristova-ontotext) [inline src/app/main/mapper/iteration/iteration.component.ts:21]: Eslint will not like this :)

File: src/app/main/mapper/mapper-dialog/mapper-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+
+        if (this.hasDatatype) {
+          const dataTypeColumnName = form.value.dataTypeColumnName;
+          const dataTypeConstant = form.value.dataTypeConstant;
+
+          if (this.isDatatypeColumn && !Helper.isBlank(dataTypeColumnName)) {
+            this.modelManagementService.setValueTypeDatatypeValueConstant(this.selected, undefined);
+            this.modelManagementService.setValueTypeDatatypeValueColumnName(this.selected, dataTypeColumnName);
+            this.modelManagementService.setValueTypeDatatypeValueSource(this.selected, form.value.dataTypeValueSource);
+            console.log('???????????????????????')
```
Comment (desislava-hristova-ontotext) [inline src/app/main/mapper/mapper-dialog/mapper-dialog.component.ts:249]: Remove this

File: src/app/models/mappable.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import {ValueTransformationImpl} from 'src/app/models/value-transformation-impl';
+import {IRIImpl} from 'src/app/models/iri-impl';
+import {ColumnImpl} from 'src/app/models/column-impl';
+import {SimpleIRIValueMappingImpl} from 'src/app/models/simple-iri-value-mapping-impl';
+import {PropertyMappingImpl} from 'src/app/models/property-mapping-impl';
+
+export interface Mappable {
```
Comment (desislava-hristova-ontotext) [inline src/app/models/mappable.ts:7]: The Mappable interface is the common type for all ValueMappings, right? So I suggest it be CellMappingBase or simply MappingBase.

Comment (teodossidossev) [reply src/app/models/mappable.ts]: changed to MappingBase

File: src/app/main/mapper/mapper-dialog/mapper-dialog.component.ts (hunk 2 above)
Comment (teodossidossev) [reply src/app/main/mapper/mapper-dialog/mapper-dialog.component.ts]: deleted

File: src/app/main/mapper/iteration/iteration.component.ts [hunk 4]
```diff
@@ -1,33 +1,93 @@
...
+  public openMapperDialog($event, item, selected) {
+    const dialogRef = this.dialog.open(MapperDialogComponent, {
+      data: {
+        mappingData: item,
+        selected,
+      },
+    });
+
+    dialogRef.afterClosed().subscribe(() => {
+      this.ngOnInit();
```
[the remark this answers is not in the document]
Comment (teodossidossev) [reply src/app/main/mapper/iteration/iteration.component.ts:90]: I know. I thought I'd fix it later, but I forgot.

File: src/app/main/mapper/mapper-dialog/mapper-dialog.component.html [hunk 5]
```diff
@@ -0,0 +1,168 @@
+<div class="subject-dialog" appCypressData="subject-dialog">
```
[the remark this answers is not in the document]
Comment (teodossidossev) [reply src/app/main/mapper/mapper-dialog/mapper-dialog.component.html:1]: I thought about it. I will do it when I apply the css

**Pull request merged**